### ISSUE TITLE
Add transport used and authentication method to result output

### ISSUE THREAD

**Issue by aaronlippold** [inspec/inspec#5690]
In our use of InSpec in the A&A process the question was asked if we ued an authentencated vs unauthenticated scan was used. 

I know we would not want to add the auth details to the results output but a middle ground could be which transport and authentication type used for the scan could be good enough.

We could add the element to the stats block and not break any existing integrations.

values could be: 
- ssh+userpass
- winrm+http
- docker+noauth
- local+localauth
- mock+noauth

**Comment by teds-mitre** [comment]
I agree that it's necessary to provide some indication of the authenticated status of an InSpec scan in its output. I also agree about _not_ including sensitive info (auth details) in the output. I'm open to the solution above and/or some equivalent to what that validator script seeks in confirming whether an ACAS (Nessus) scan is credentialed. For the values, suggest we be explicit that the first two are authenticated: ssh+userpass auth, winrm+https auth. Redundant, but substantial business benefit from denoting these as valid auth mechanisms.
